pandas/core/arrays/timedeltas.py

```diff
@@ -678,6 +754,30 @@ def to_pytimedelta(self):
         """
         return tslibs.ints_to_pytimedelta(self.asi8)
 
+    def astype(self, dtype, copy=True):
+        # We handle
+        # --> timedelta64[ns]
+        # --> timedelta64
+        dtype = pandas_dtype(dtype)
+
+        if is_timedelta64_dtype(dtype) and not is_timedelta64_ns_dtype(dtype):
+            # essentially this is division
+            result = self._data.astype(dtype, copy=copy)
+            if self._hasnans:
+                values = self._maybe_mask_results(result,
+                                                  fill_value=None,
+                                                  convert='float64')
+                return values
+            return result.astype('i8')
```

Comment:
why returning an integer dtype here?   id guess releated to "essentially this is division" comment, but still seems weird to me.

Reply:
This is unchanged from TimedeltaIndex.astype on master.

```
        if is_timedelta64_dtype(dtype) and not is_timedelta64_ns_dtype(dtype):
            # return an index (essentially this is division)
            result = self.values.astype(dtype, copy=copy)
            if self.hasnans:
                values = self._maybe_mask_results(result, fill_value=None,
                                                  convert='float64')
                return Index(values, name=self.name)
            return Index(result.astype('i8'), name=self.name)
 ```

Reply:
For those following along, I think this conversation has splintered between here and https://github.com/pandas-dev/pandas/pull/24405